File: tailingZero.rb
def factorial(a) 
   if a < 2 
      return 1 
      end 
    return a*factorial(a-1)
    end 
def tailingZero(a) 
   fact = factorial(a)
   count = 0
   while fact%10 == 0 do 
      count = count+1
      fact = fact/10 
      end 
    return count 
    end 
str = gets.chomp
a = str.to_i 
puts tailingZero(a)

File: TailingZero.exs
defmodule TailingZeroProblem do
  def factorial(n) do
    if n < 2 do
      1
    else
      n*factorial(n-1)
    end
  end
  def tailingZero(n), do: countZero(n,0)
   defp countZero(n,count) do
    if rem(n,10) != 0 do
      count
    else
      countZero(div(n,10),count+1)
    end
   end
end
IO.puts 5 |> TailingZeroProblem.factorial() |> TailingZeroProblem.tailingZero()
IO.puts 2 |> TailingZeroProblem.factorial() |> TailingZeroProblem.tailingZero()
IO.puts 10 |> TailingZeroProblem.factorial() |> TailingZeroProblem.tailingZero()
IO.puts 43|> TailingZeroProblem.factorial() |> TailingZeroProblem.tailingZero()
IO.puts 21 |> TailingZeroProblem.factorial() |> TailingZeroProblem.tailingZero()
